src/SFA.DAS.EAS.Employer_Financial.Database/Scripts/PreDeployment/AML-2873_TransferPaymentDate.sql

```diff
@@ -0,0 +1,61 @@
+﻿IF COL_LENGTH('employer_financial.TransactionLine', 'Id') IS NULL
+BEGIN
+BEGIN TRY
+BEGIN TRANSACTION
+    CREATE TABLE [employer_financial].[TransactionLine2]
+	(
+		Id BIGINT PRIMARY KEY IDENTITY(1,1) NOT NULL,
+		AccountId BIGINT NOT NULL,
+		DateCreated DATETIME NOT NULL,
+		SubmissionId BIGINT NULL,
+		TransactionDate DATETIME NOT NULL,
+		TransactionType TINYINT NOT NULL DEFAULT 0, 
+		LevyDeclared DECIMAL(18,4) NULL, 
+		Amount DECIMAL(18,4) NOT NULL DEFAULT 0, 
+		EmpRef nVarchar(50) null,
+		PeriodEnd nVarchar(50) null,
+		Ukprn BIGINT null, 
+		SfaCoInvestmentAmount DECIMAL(18, 4) NOT NULL DEFAULT 0, 
+		EmployerCoInvestmentAmount DECIMAL(18, 4) NOT NULL DEFAULT 0,
+		[EnglishFraction] DECIMAL(18, 5) NULL, 
+		[TransferSenderAccountId] BIGINT NULL, 
+		[TransferSenderAccountName] NVARCHAR(100) NULL,
+		[TransferReceiverAccountId] BIGINT NULL, 
+		[TransferReceiverAccountName] NVARCHAR(100) NULL
+	)
+	
+
+INSERT INTO [employer_financial].[TransactionLine2] SELECT * FROM [employer_financial].[TransactionLine]
+
+
+EXEC sp_rename 'employer_financial.TransactionLine', 'TransactionLine3'
+EXEC sp_rename 'employer_financial.TransactionLine2', 'TransactionLine'
+
+
+CREATE INDEX [IX_TransactionLine_SubmissionId] ON [employer_financial].[TransactionLine] (SubmissionId)
+CREATE INDEX [IX_TransactionLine_AccountId] ON [employer_financial].[TransactionLine] (AccountId) INCLUDE (Ukprn,PeriodEnd,TransactionType)
+CREATE INDEX [IX_TransactionLine_Payment] on [employer_financial].[TransactionLine] (PeriodEnd,AccountId,Ukprn,TransactionDate, DateCreated)
+CREATE UNIQUE INDEX [IX_TransactionLine_TransactionType_SubmissionId] ON [employer_financial].[TransactionLine] (SubmissionId) WHERE (TransactionType = 1);
+CREATE INDEX [IX_TransactionLine_AccountId_DateCreated] ON [employer_financial].[TransactionLine] (AccountId, DateCreated) WITH (ONLINE = ON)
+CREATE INDEX [IX_TransactionLine_Account_TransactionType] ON [employer_financial].[TransactionLine] (AccountId, TransactionType) INCLUDE (DateCreated) WITH (ONLINE = ON)
+
+
+DROP TABLE employer_financial.TransactionLine3
+
+END TRY
+BEGIN CATCH
+DECLARE @ErrorMsg nvarchar(max)
+DECLARE @ErrorSeverity INT;
+DECLARE @ErrorState INT;
+SET @ErrorMsg = ERROR_NUMBER() + ERROR_LINE() + ERROR_MESSAGE()
+SET @ErrorSeverity = ERROR_SEVERITY()
+SET @ErrorState = ERROR_STATE()
+
+IF @@TRANCOUNT > 0
+    ROLLBACK TRANSACTION
+
+RAISERROR(@ErrorMsg, @ErrorSeverity, @ErrorState)
+END CATCH
+IF @@TRANCOUNT > 0
+ COMMIT TRANSACTION
```

Comment:
The transaction is created within the scope of the BEGIN TRY, so it would seem natural to commit it within the same scope...
i.e. BEGIN TRY >  BEGIN TRAN > stuff > COMMIT > END TRY
 rather than:
 BEGIN TRY >  BEGIN TRAN > stuff > END TRY > COMMIT